src/dbnode/storage/index/aggregate_results.go

```diff
@@ -62,9 +129,14 @@ func NewAggregateResults(
 	aggregateOpts AggregateResultsOptions,
 	opts Options,
 ) AggregateResults {
+	if aggregateOpts.AggregateUsageMetrics == nil {
+		aggregateOpts.AggregateUsageMetrics = &usageMetrics{}
+	}
```

Comment:
Shouldn't this just return error if not set?  Don't see why we should support this being nil and making no-op metrics, that is the kind of pattern which ends up with us accidentally setting nil here and then not catching the issue until we find out the metrics aren't there and we trace why the metrics are missing (instead of actually have a failing build in CI or other kind of testing).